Sources/VGSShowSDK/UIElements/VGSLabel/VGSLabel+Internal.swift

```diff
@@ -166,79 +168,79 @@ internal extension VGSLabel {
-
-		let formattingEvent = "Text before formatting: \"\(text)\", text after formatting: \"\(maskedText)\" ."
-		logInfoEventWithText(formattingEvent)
-
-    if isSecureText {
-      let securedText = secureTextInRanges(maskedText, ranges: secureTextRanges)
-      updateMaskedLabel(with: securedText)
-
-			let eventText = "Apply custom formatting. Apply secure masks."
-			logInfoEventWithText(eventText)
-
-      // Set masked text as accessibility value when applyting secure formatting.
-      label.accessibilityValue = maskedText
-      return
+    /// Update text and apply transformation regex if available.
+    func updateTextAndMaskIfNeeded() {
+        // Mask only normal text.
+        guard let text = revealedRawText else {
+            label.secureText = nil
+            
+            // Reset accessibility value.
+            vgsAccessibilityValue = nil
+            // No revealed text - show placeholder, hide main text label.
+            label.isHidden = true
+            updatePlaceholder()
+            
+            logPlaceholderEvent(isShown: true)
+            return
+        }
+        
+        // Hide placeholder.
+        placeholderLabel.isHidden = true
+        
+        // Unhide main label.
+        label.isHidden = false
+        
+        logPlaceholderEvent(isShown: false)
+        
+        // No mask: set revealed text.
+        guard textFormattersContainer.hasFormatting else {
+            
+            // Set raw text to accessibility value (no formatting is applied)
+            vgsAccessibilityValue = revealedRawText
+            // Check if text should be secured.
+            if isSecureText {
+                let securedText = secureTextInRanges(text, ranges: secureTextRanges)
+                updateMaskedLabel(with: securedText)
+                
+                let eventText = "No custom formatting. Apply secure mask for revealed data."
+                logInfoEventWithText(eventText)
+                
+                return
+            }
+            
+            let eventText = "No custom formatting. No secure masks. Show raw revealed data."
+            logInfoEventWithText(eventText)
+            
+            updateMaskedLabel(with: text)
+            return
+        }
+        
+        // Set masked text to label.
+        let maskedText = textFormattersContainer.formatText(text)
+        
+        let formattingEvent = "Text before formatting: \"\(text)\", text after formatting: \"\(maskedText)\" ."
+        logInfoEventWithText(formattingEvent)
+        
+        if isSecureText {
+            let securedText = secureTextInRanges(maskedText, ranges: secureTextRanges)
+            updateMaskedLabel(with: securedText)
+            
+            let eventText = "Apply custom formatting. Apply secure masks."
+            logInfoEventWithText(eventText)
+            
+            // Set masked text as accessibility value when applyting secure formatting.
+            vgsAccessibilityValue = maskedText
```

Comment:
replace **vgs** prefix, probably it can be **labelAccessibilityValue** similar to Collect SDK

Reply:
These two properties were already created before we do the accessibility improvements:
- `vgsAccessibilityLabel`
- `vgsAccessibilityHint`

I think we cannot change those values since it will be a breaking change?

The new property is:
`vgsIsAccessibilityElement`

I'm not sure if we should remove the vgs prefix from it since it is like a aggregation to the current accessibility implementation. What do you think?

Reply:
ok, lets keep vgs prefic then